src/hooks.ts

```diff
@@ -261,3 +258,129 @@ export function useConvexPolyhedron(fn: ConvexPolyhedronFn, deps: any[] = []) {
+  axisA?: [number, number, number]
+  pivotB?: [number, number, number]
+  axisB?: [number, number, number]
+}
+type DistanceConstraintOpts = ConstraintOptns & { distance?: number }
+
+type HingeConstraintOpts = ConstraintOptns & {
+  pivotA?: [number, number, number]
+  axisA?: [number, number, number]
+  pivotB?: [number, number, number]
+  axisB?: [number, number, number]
+}
+
+type LockConstraintOpts = ConstraintOptns & {}
+
+export function usePointToPointConstraint(
+  bodyA: React.MutableRefObject<THREE.Object3D | undefined>,
+  bodyB: React.MutableRefObject<THREE.Object3D | undefined>,
+  optns: PointToPointConstraintOpts,
+  deps: any[]
+) {
+  return useConstraint('PointToPoint', bodyA, bodyB, optns, deps)
+}
+export function useConeTwistConstraint(
+  bodyA: React.MutableRefObject<THREE.Object3D | undefined>,
+  bodyB: React.MutableRefObject<THREE.Object3D | undefined>,
+  optns: ConeTwistConstraintOpts,
+  deps: any[]
+) {
+  return useConstraint('ConeTwist', bodyA, bodyB, optns, deps)
+}
+export function useDistanceConstraint(
+  bodyA: React.MutableRefObject<THREE.Object3D | undefined>,
+  bodyB: React.MutableRefObject<THREE.Object3D | undefined>,
+  optns: DistanceConstraintOpts,
+  deps: any[]
+) {
+  return useConstraint('Distance', bodyA, bodyB, optns, deps)
+}
+export function useHingeConstraint(
+  bodyA: React.MutableRefObject<THREE.Object3D | undefined>,
+  bodyB: React.MutableRefObject<THREE.Object3D | undefined>,
+  optns: HingeConstraintOpts,
+  deps: any[]
+) {
+  return useConstraint('Hinge', bodyA, bodyB, optns, deps)
+}
+export function useLockConstraint(
+  bodyA: React.MutableRefObject<THREE.Object3D | undefined>,
+  bodyB: React.MutableRefObject<THREE.Object3D | undefined>,
+  optns: LockConstraintOpts,
+  deps: any[]
+) {
+  return useConstraint('Lock', bodyA, bodyB, optns, deps)
+}
+
+function useConstraint(
+  type: ConstraintTypes,
+  bodyA: React.MutableRefObject<THREE.Object3D | undefined>,
+  bodyB: React.MutableRefObject<THREE.Object3D | undefined>,
+  optns: any = {},
+  deps: any[] = []
+) {
+  const { worker } = useContext(context)
+  const uuid = THREE.MathUtils.generateUUID()
+
+  useEffect(() => {
+    if (bodyA.current && bodyB.current) {
+      worker.postMessage({
+        op: 'addConstraint',
+        uuid: uuid,
+        type: type,
+        props: [bodyA.current.uuid, bodyB.current.uuid, optns],
+      })
+
+      return () =>
+        worker.postMessage({
+          op: 'removeConstraint',
+          uuid: uuid,
+        })
+    }
+  }, deps)
+}
+
+export function useSpring(
+  bodyA: React.MutableRefObject<THREE.Object3D | undefined>,
+  bodyB: React.MutableRefObject<THREE.Object3D | undefined>,
+  optns: any = {},
+  deps: any[] = []
+) {
+  const { worker, events } = useContext(context)
+  const uuid = THREE.MathUtils.generateUUID()
+
+  useEffect(() => {
+    if (bodyA.current && bodyB.current) {
+      worker.postMessage({
+        op: 'addSpring',
+        uuid: uuid,
```

Comment:
This can just be `uuid,` (instead of `uuid: uuid,`) if the object key has the same name as the value